bpo-35436: Add missing PyErr_NoMemory() calls and other minor bug fixes

ZackerySpytz opened the pull request:
Set MemoryError when appropriate, add missing failure checks,
and fix some potential leaks.

<!-- issue-number: [bpo-35436](https://bugs.python.org/issue35436) -->
https://bugs.python.org/issue35436
<!-- /issue-number -->

ZackerySpytz committed:
bpo-35436: Add missing PyErr_NoMemory() calls and other minor bug fixes

Set MemoryError when appropriate, add missing failure checks,
and fix some potential leaks.

serhiy-storchaka commented on Parser/myreadline.c:
```diff
@@ -211,17 +228,23 @@ PyOS_StdioReadline(FILE *sys_stdin, FILE *sys_stdout, const char *prompt)
+                            PyErr_NoMemory();
+                            return NULL;
+                        }
+                        else {
```
You can remove `else` and decrease an indentation level.

serhiy-storchaka reviewed (approved):
Great work! It is worth adding a news entry.

ZackerySpytz committed:
Add a news entry

Remove some useless indentation.
Add a missing check introduced in a recent commit (468a15a).
